src/js/player.js

```diff
@@ -1448,21 +1450,24 @@ class Player extends Component {
         return;
       }
 
-      return mutedPromise.catch(restoreMuted);
+      return mutedPromise.catch(err => {
+        restoreMuted();
+        throw new Error(`Rejection at manualAutoplay. Restoring muted value. ${err ? err : ''}`);
```

Comment:
If I remove this line that throws the error, I would expect [these tests](https://github.com/videojs/video.js/blob/d2241ca64f7498f71608d3ec36739f62e593f199/test/unit/autoplay.test.js#L278-L319) to fail since the promise on line [1479](https://github.com/videojs/video.js/pull/7293/files#diff-f92dc431786299ba52ba41d6901e029cc1eeb2f5603653bd03eb4d88a27f5020R1479) should resolve and result in an `'autoplay-success'` event and increment `this.counts.success` rather than `this.counts.failure`, however the test still passes which suggests it isn't capturing the test case exactly. It might be necessary to modify `this.rejectPromise` for these tests. I've shared one idea below but I'm not positive it will work.

Reply:
Maybe what we need is a separate promise fake for these two tests?

Reply:
I tried it locally, and I think it might be too complicated to get it in now.

Reply:
In that case how about adding a `TODO` comment to fix these two tests at some point?

Reply:
Spent some time playing around with sinon's promises, and probably needs a complete rewrite of the test to use that.
I'll create an issue for this.

Reply:
#7295